11embedding/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+)
+
+// matcher defines the behavior required for performing matching.
+type matcher interface {
+	match(searchTerm string) bool
+}
+
+// sport represents a sports team.
+type sport struct {
+	team string
+	city string
+}
+
+// match checks the value for the specified term.
+func (s sport) match(searchTerm string) bool {
+	return strings.Contains(s.team, searchTerm) || strings.Contains(s.city, searchTerm)
+}
+
+// Declare a struct type named hockey that represents specific
+// hockey information. Have it embed the sport type first.
+
+type hockey struct {
+	sport
+	grade string
+}
+
+// match checks the value for the specified term.
+func (h hockey) match(searchTerm string) bool {
+
+	return h.sport.match(searchTerm) || strings.Contains(h.grade, searchTerm)
+
+	// Make sure you call into match method for the embedded sport type.
```

Comment:
Why is your comment below? Either move it above or put in into the function documentation.